### PR TITLE
[Draft] Support DAL model

### PR DESCRIPTION
**Pull request by heiyuxiaokai**
Reproduce codes in mmrotate of DAL——《Dynamic Anchor Learning for Arbitrary-Oriented Object Detection》
paper: https://arxiv.org/abs/2012.04150
origin codes: https://github.com/ming71/DAL
## Motivation

There are still some performance gaps with origin codes:
1. Performance on the train-val set is lower than in other methods.
2. Performance on the test set is lower than given in the paper.

**Results on train-val:**
| method | schedule | bs | data aug |   mAP    |
| :---: | :---: | :---: | :---: | :---: |
|    DAL    |      1x       | 2 |     -        |  68.77  |
|    DAL    |      2x       | 2 |   -        |  75.5  |
|    DAL    |      3x       | 2 |    -        |  77.7  |
|    DAL    |      3x       | 2 |   rr        |  76.5  |
|    DAL    |      3x       | 2 |   ms+rr   |  -  |

**Results on test:**
| method | schedule | bs | data aug |   mAP    |
| :---: | :---: | :---: | :---: | :---: |
|    DAL    |      3x       | 2 |     -        |  69.21  |
|    DAL    |      3x       | 2 |   rr        |  69.99  |

**It would be great if someone could give me some advice.**

## Modification

- [x] support DAL
- [ ] Benchmark the performance
- [ ] Prepare and release the models
- [ ] Prepare a docstrings

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
4. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
5. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by zytx121** [comment]
Thank you for your PR, it's awesome!

**Comment by jbwang1997** [comment]
Thank you very much for your great contributions. As described in the original paper, 30k iterations have been applied for training DAL on DOTA dataset. random flip, rotation, and HSV color space transformation for data augmentation. What's more, the image splitting strategy is also different from the default setting in mmrotate, which is 800 x 800 with 200 overlaps. You can first align the training set with the DAL code.
